### PR TITLE
Add launch script for gem5-20 boot tests

### PR DESCRIPTION
**Pull request by ayaz91**

### CODE REVIEW

File: launch-scripts/launch_boot_tests_gem5_20.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    num_cpus = ['1', '2', '4', '8']
+    cpu_types = ['kvm', 'atomic', 'simple', 'o3']
+    mem_types = ['classic', 'MI_example', 'MESI_Two_Level']
+
+    for linux in linuxes:
+        for boot_type in boot_types:
+            for cpu in cpu_types:
+                for num_cpu in num_cpus:
+                    for mem in mem_types:
+                        if mem == 'MESI_Two_Level':
```
Comment (powerjg) [inline launch-scripts/launch_boot_tests_gem5_20.py:128]: Why is MESI_Two_Level different?

Comment (ayaz91) [reply launch-scripts/launch_boot_tests_gem5_20.py:128]: The binary location (and binary artifacts) are different.

**Comment by ayaz91** [comment]
@powerjg: please, have a look at this, whenever you get a chance

File: launch-scripts/launch_boot_tests_gem5_20.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        else:
+            binary_gem5 = 'gem5/build/X86/gem5.opt'
+            artifact_gem5 = gem5_binary
+
+        return gem5Run.createFSRun(
+            'boot experiments with gem5-20',
+            binary_gem5,
+            'configs-boot-tests/run_exit.py',
+            'results/run_exit/vmlinux-{}/boot-exit/{}/{}/{}/{}'.
+            format(linux, cpu, mem, num_cpu, boot_type),
```
Comment (powerjg) [inline launch-scripts/launch_boot_tests_gem5_20.py:139]: Can you add a few spaces of indentation here?
```suggestion
                format(linux, cpu, mem, num_cpu, boot_type),
```

File: launch-scripts/launch_boot_tests_gem5_20.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+            artifact_gem5, gem5_repo, experiments_repo,
+            os.path.join('linux-stable', 'vmlinux'+'-'+linux),
+            'disk-image/boot-exit/boot-exit-image/boot-exit',
+            linux_binaries[linux], disk_image,
+            cpu, mem, num_cpu, boot_type,
+            timeout = 6*60*60 #6 hours
+            )
+
+    for run in starmap(createRun, product(linuxes, boot_types, cpu_types, num_cpus, mem_types)):
```
Comment (powerjg) [inline launch-scripts/launch_boot_tests_gem5_20.py:148]: Let's split this into a couple of lines to make it a bit more clear.

```suggestion
    # For the cross product of tests, create a run object.
    runs = starmap(createRun, product(linuxes, boot_types, cpu_types, num_cpus, mem_types)):
    # Run all of these experiments in parallel
    for run in runs:
```

**Comment by ayaz91** [comment]
"commit suggestion" is  a cool feature

**Comment by powerjg** [comment]
Yeah, much easier than you having to update the commit :).

It looks good to me, now! Thanks for all of the updates!

**Pull request merged**